proto/data.go

```diff
@@ -96,7 +96,14 @@ func (k Key) Next() Key {
 	return Key(bytesNext(k))
 }
 
+// IsPrev is a more efficient version of k.Next().Equal(m).
+func (k Key) IsPrev(m Key) bool {
+	l := len(m) - 1
+	return l == len(k) && m[l] == 0
```

Comment:
This isn't enough. We still need to check that `k` and `m[:l]` are equal (consider `foo` and `bar\x00`).